src/tests/end-to-end/binding_constraints/test_binding_constraints.cpp

```diff
@@ -2,548 +2,276 @@
 
-    // Binding constraint parameter varying depending on the test
-    BC->mutateTypeWithoutCheck(BindingConstraint::typeHourly);
+    BC->setTimeGranularity(BindingConstraint::typeHourly);
     BC->operatorType(BindingConstraint::opLess);
 
-    unsigned int numberOfTS = 1;
-    BCrhsConfig bcRHSconfig(BC, numberOfTS);
-
     double rhsValue = 90.;
-    bcRHSconfig.fillTimeSeriesWith(0, rhsValue);
-
-    BCgroupScenarioBuilder bcGroupScenarioBuilder(study, nbYears);
-    bcGroupScenarioBuilder.yearGetsTSnumber(BC->group(), 0, 0);
+    TimeSeriesConfigurer(BC->RHSTimeSeries())
+        .setColumnCount(1)
+        .fillColumnWith(0, rhsValue);
 
     simulation->create();
-    giveWeigthOnlyToYear(0);
     simulation->run();
 
-    unsigned int hour = 100;
-    BOOST_TEST(getLinkFlowAthour(simulation->get(), link, hour) <= rhsValue, tt::tolerance(0.001));
+    BOOST_TEST(output->flow(link).hour(100) <= rhsValue, tt::tolerance(0.001));
 }
 
 BOOST_AUTO_TEST_CASE(Daily_BC_restricts_link_direct_capacity_to_greater_than_80)
 {
-    // Study parameters varying depending on the test
-    unsigned int nbYears = 1;
-    setNumberMCyears(study, nbYears);
+    setNumberMCyears(1);
 
-    // Binding constraint parameter varying depending on the test
-    BC->mutateTypeWithoutCheck(BindingConstraint::typeDaily);
+    BC->setTimeGranularity(BindingConstraint::typeDaily);
     BC->operatorType(BindingConstraint::opGreater);
 
-    unsigned int numberOfTS = 1;
-    BCrhsConfig bcRHSconfig(BC, numberOfTS);
-
     double rhsValue = 80.;
-    bcRHSconfig.fillTimeSeriesWith(0, rhsValue);
+    TimeSeriesConfigurer(BC->RHSTimeSeries())
+        .setColumnCount(1)
+        .fillColumnWith(0, rhsValue);
 
-    BCgroupScenarioBuilder bcGroupScenarioBuilder(study, nbYears);
-    bcGroupScenarioBuilder.yearGetsTSnumber(BC->group(), 0, 0);
+    simulation->create();
+    simulation->run();
+
+    BOOST_TEST(output->flow(link).hour(100) >= rhsValue, tt::tolerance(0.001));
+}
+
+BOOST_AUTO_TEST_SUITE_END()
+
+
+BOOST_FIXTURE_TEST_SUITE(TESTS_BINDING_CONSTRAINTS_ON_A_CLUSTER, StudyWithBConCluster)
+
+BOOST_AUTO_TEST_CASE(Hourly_BC_restricts_cluster_generation_to_90)
+{
+    setNumberMCyears(1);
+
+    BC->setTimeGranularity(BindingConstraint::typeHourly);
+    BC->operatorType(BindingConstraint::opEquality);
+
+    double rhsValue = 90.;
+    TimeSeriesConfigurer(BC->RHSTimeSeries())
+        .setColumnCount(1)
+        .fillColumnWith(0, rhsValue);
 
     simulation->create();
-    giveWeigthOnlyToYear(0);
     simulation->run();
 
-    unsigned int hour = 100;
-    BOOST_TEST(getLinkFlowAthour(simulation->get(), link, hour) >= rhsValue, tt::tolerance(0.001));
+    BOOST_TEST(output->thermalGeneration(cluster.get()).hour(10) == rhsValue, tt::tolerance(0.001));
 }
 
 BOOST_AUTO_TEST_SUITE_END()
 
 
-BOOST_FIXTURE_TEST_SUITE(TESTS_ON_BC_RHS_SCENARIZATION, Fixture)
+BOOST_FIXTURE_TEST_SUITE(TESTING_BC_RHS_SCENARIZATION_WHEN_BC_ON_A_LINK, StudyWithBConLink)
```

Comment:
Please keep using `BOOST_FIXTURE_TEST_SUITE`, so that `Fixture::Fixture` and `Fixture::~Fixture` will be called for each test.